File: core/src/main/java/com/canyon/commons/MD5Utils.java
package com.canyon.commons;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;

public class MD5Utils {
	public static String getMD5fromFile(String path) {
		String result = "";
		if (IOUtils.isFile(path))
			return result;

		FileInputStream stream = null;
		try {
			stream = new FileInputStream(path);
			result = org.apache.commons.codec.digest.DigestUtils.md5Hex(stream);
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			e.printStackTrace();
		} finally {
			if (stream != null) {
				try {
					stream.close();
				} catch (IOException e) {
					e.printStackTrace();
				}
			}
		}
		return result;
	}

	public static String getMD5fromString(String text){
		return org.apache.commons.codec.digest.DigestUtils.md5Hex(text);
	}
}

File: core/src/main/kotlin/com/canyon/commons/MD5Utils.kt
package com.canyon.commons

import java.io.File
import java.io.FileInputStream

object MD5Utils {
    fun getMD5fromFile(path: String): String {
        var file = File(path)
        if (file.isFile && file.canRead()) {
            FileInputStream(path).use { fos ->
                return org.apache.commons.codec.digest.DigestUtils.md5Hex(fos)
            }
        }
        throw IllegalArgumentException("The file can't read or not exists. path:$path")
    }

    fun getMD5fromString(text: String): String {
        return org.apache.commons.codec.digest.DigestUtils.md5Hex(text)
    }
}
